scipy/integrate/_ivp/ivp.py

```diff
@@ -628,9 +626,9 @@ def fun(t, x, fun=fun):
 
     interpolants = []
 
-    events, is_terminal, event_dir = prepare_events(events)
-
     if events is not None:
+        events, max_events, event_dir = prepare_events(events)
+        event_count = np.zeros(len(events))
```

Comment:
Not that it really matters but if there are other changes to be made
```suggestion
        event_count = np.zeros_like(events)
```

Reply:
I think this would give it object dtype, and I don't think we want that.